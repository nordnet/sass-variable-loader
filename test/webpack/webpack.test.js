/* eslint-disable import/no-extraneous-dependencies */
const webpack = require('webpack');
const path = require('path');
const rm = require('rimraf');

const config = {
  entry: './main.js',
  context: __dirname,
  output: {
    path: path.join(__dirname, 'dist'),
    filename: 'variables.js',
    libraryTarget: 'commonjs2',
  },
};

function cleanDistDirectory() {
  return new Promise((resolve, reject) => {
    rm(config.output.path, err => {
      if (err) {
        reject(err);
        return;
      }
      resolve();
    });
  });
}

function compile() {
  return new Promise((resolve, reject) => {
    webpack(config).run(err => {
      if (err) {
        reject(err);
        return;
      }
      resolve();
    });
  });
}

function testFn() {
  test('variables should be a none-empty object', async () => {
    try {
      await cleanDistDirectory();
      await compile();
    } catch (error) {
      throw error;
    }
    // eslint-disable-next-line global-require, import/no-unresolved
    const variables = require('./dist/variables');
    expect(typeof variables).toBe('object');
    expect(Object.keys(variables).length).toBeGreaterThan(20);
  });
}

describe('webpack', testFn);
