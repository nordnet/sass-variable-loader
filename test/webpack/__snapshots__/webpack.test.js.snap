// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`webpack bulma variables should match snapshot 1`] = `
Object {
  "black": "#0a0a0a",
  "black-bis": "#121212",
  "black-ter": "#242424",
  "blue": "#3273dc",
  "cyan": "#209cee",
  "desktop": "1024px",
  "easing": "ease-out",
  "family-monospace": "monospace",
  "family-sans-serif": "BlinkMacSystemFont, -apple-system, \\"Segoe UI\\", \\"Roboto\\", \\"Oxygen\\", \\"Ubuntu\\", \\"Cantarell\\", \\"Fira Sans\\", \\"Droid Sans\\", \\"Helvetica Neue\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
  "fullhd": "1408px",
  "gap": "32px",
  "green": "#23d160",
  "grey": "#7a7a7a",
  "grey-dark": "#4a4a4a",
  "grey-darker": "#363636",
  "grey-light": "#b5b5b5",
  "grey-lighter": "#dbdbdb",
  "orange": "#ff470f",
  "purple": "#b86bff",
  "radius": "3px",
  "radius-large": "5px",
  "radius-rounded": "290486px",
  "radius-small": "2px",
  "red": "#ff3860",
  "render-mode": "optimizeLegibility",
  "size-1": "3rem",
  "size-2": "2.5rem",
  "size-3": "2rem",
  "size-4": "1.5rem",
  "size-5": "1.25rem",
  "size-6": "1rem",
  "size-7": "0.75rem",
  "speed": "86ms",
  "tablet": "769px",
  "turquoise": "#00d1b2",
  "variable-columns": "true",
  "weight-bold": "700",
  "weight-light": "300",
  "weight-medium": "500",
  "weight-normal": "400",
  "weight-semibold": "600",
  "white": "white",
  "white-bis": "#fafafa",
  "white-ter": "whitesmoke",
  "widescreen": "1216px",
  "yellow": "#ffdd57",
}
`;
